plugins/scanners/scancode/src/main/kotlin/ScanCodeResultModelMapper.kt

```diff
@@ -85,12 +85,20 @@ fun ScanCodeResult.toScanSummary(preferFileLicense: Boolean = false): ScanSummar
 
     filesOfTypeFile.forEach { file ->
         val licensesWithoutReferences = file.licenses.filter {
-            it !is LicenseEntry.Version3 || it.fromFile == null
+            // Determine `fromFile`, which we have starting from version 3
+            var fromFile: String? = null
+            if (it is LicenseEntry.Version3) {
+                fromFile = it.fromFile
+            } else if (it is LicenseEntry.Version4) {
+                fromFile = it.fromFile
+            }
```

Comment:
Should we turn this into a `when` clause, so that we get remembered by a complaint about being non exhaustive once `LicenseEntry.Version5` gets introduced?

Reply:
I was just going to fix my commit... this seems to work fine and looks much better.
```kotlin
// Determine `fromFile`, which we have starting from version 3
var fromFile: String? = when(it) {
    is LicenseEntry.Version1 -> null
    is LicenseEntry.Version3 -> it.fromFile
    is LicenseEntry.Version4 -> it.fromFile
}
```

@sschuberth Unfortunately, I am very new to kotlin and just learned it to fix this problem. Especially I do not understand the serialization logic, so I do not think that I am able to change the interface in a way that works with all versions.

Reply:
I'm believe the commit makes sense as-is, and further improvements can be made on top (by someone else possibly).

Reply:
I've very much like to not merge the code as-is, but improve it already now, as IMO the current implementation (while solving the problem) goes into the wrong direction "software architectural" wise. I guess any follow-up would just revert this and do it differently. In that sense we should do it right from the start.

I can offer some guidance here, but not right now, as I'm on the road.

Reply:
sure @sschuberth - I'll leave that PR up to you then.